src/common/service/NetRemoteService.cxx

```diff
@@ -389,6 +399,125 @@ NetRemoteService::WifiAccessPointDisable([[maybe_unused]] ::grpc::ServerContext*
     return grpc::Status::OK;
 }
 
+using Microsoft::Net::Wifi::Dot11PhyType;
+
+::grpc::Status
+NetRemoteService::WifiAccessPointSetPhyType([[maybe_unused]] ::grpc::ServerContext* context, const ::Microsoft::Net::Remote::Wifi::WifiAccessPointSetPhyTypeRequest* request, ::Microsoft::Net::Remote::Wifi::WifiAccessPointSetPhyTypeResult* response)
+{
+    LOGD << std::format("Received WifiAccessPointSetPhyType request for access point id {}", request->accesspointid());
+
+    WifiAccessPointOperationStatus status{};
+    status.set_code(WifiAccessPointOperationStatusCode::WifiAccessPointOperationStatusCodeSucceeded);
+
+    if (request->phytype() == Dot11PhyType::Dot11PhyTypeUnknown) {
+        status.set_code(WifiAccessPointOperationStatusCode::WifiAccessPointOperationStatusCodeInvalidParameter);
+        status.set_message("No PHY type provided");
+    } else {
+        auto accessPointWeak = m_accessPointManager->GetAccessPoint(request->accesspointid());
+        auto accessPointController = detail::IAccessPointWeakToAccessPointController(accessPointWeak);
+        if (!accessPointController) {
+            LOGE << std::format("Failed to create controller for access point {}", request->accesspointid());
+            status.set_code(WifiAccessPointOperationStatusCode::WifiAccessPointOperationStatusCodeAccessPointInvalid);
+            status.set_message(std::format("Failed to create controller for access point {}", request->accesspointid()));
+        }
+
+        if (!accessPointController->SetPhyType(request->phytype())) {

```

Comment:
One thing (I think) we need to account for is whether the specified phy type is supported, where in the stack to perform this check, and how to signal any errors to the API client related to arguments that aren't supported. Currently IAccessPointController::SetPhyType() returns a `bool` which I think is fine, but that means the check for support will need to be done at a higher level since the `bool` return type can't provide any detailed information beyond success/fail.

This information is in the `Ieee80211AccessPointCapabilities` object, which can be obtained using `IAccessPointController::GetCapabilities()`, so the API implementation could check this directly.

I suppose an alternative is to not do the check at all, and just let the lowest layer fail on bad/invalid arguments. My only concern is that the caller then can't differentiate between the cases where the failure occurred due to some temporary problem, or whether the arguments were truly unsupported. That's why I lean toward adding an explicit check for support against the requested phy type.

This pattern will apply to other APIs that are setting/changing configuration values, so we should think deeply about how best to handle this and apply the solution to all of those APIs.

Reply:
Good ideas. In general, I am a fan of identifying issues as early as possible to more easily identify when the issue occurs. So, I'm leaning toward checking `Ieee80211AccessPointCapabilities` in the API implementation directly (`NetRemoteService::WifiAccessPointSetPhyType`) so that an informative status message can explain that the PHY type is not supported. Then if `AccessPointControllerLinux::SetPhyType` still returns `false`, then we know there was an issue with actually setting the PHY type value

Reply:
Sounds great.